### ISSUE TITLE
Error when installing KMCP

### ISSUE THREAD

**Issue by davidmaimoun** [shenwei356/kmcp#16]
Hi Dr Chen,
You helped me and advised me to try KMCP (biostars forum).
Sorry to disturb you again
When I try to install it via conda, it throw an error:

_What happened?
The initial connection between Cloudflare's network and the origin web server timed out. As a result, the web page can not be displayed.
If you're the owner of this website:
Contact your hosting provider letting them know your web server is not completing requests. An Error 522 means that the request was able to connect to your web server, but that the request didn't finish. The most likely cause is that something on your server is hogging resources.
https://support.cloudflare.com/hc/en-us/articles/200171906-Error-522_

Do you think we could fix it?

Thank you Dr

**Comment by davidmaimoun** [comment]
Thank you very much again Dr Shen!

**Comment by shenwei356** [comment]
The docs may help:

- Building custom databases: https://bioinf.shenwei.me/kmcp/database/#building-custom-databases
- Sequence searching https://bioinf.shenwei.me/kmcp/tutorial/searching/#sequence-search

### The 661K dataset

Index/Database Building

    # cobs ---------------------------------
    
    # 10h
    # index file: 873G
    time cobs compact-construct --file-type fasta Assemblies/ ena-bact-661k.cobs_compact  --clobber
    
    # kmcp ---------------------------------
    
    # ~24h
    # tmp files: ~19T
    kmcp compute -e -I Assemblies/ -O ena-bact-661k.kmcp-k31 --log ena-bact-661k.kmcp-k31.log  
    
    # 11h31m
    # index file 843.02 GB
    kmcp index -I ena-bact-661k.kmcp-k31/ -O ena-bact-661k.kmcp-k31.db --log ena-bact-661k.kmcp-k31.db.log

**Comment by shenwei356** [comment]
You may try other de bruijn graph based tools for highly redundant datasets, including [cuttlefish 2](https://github.com/COMBINE-lab/cuttlefish), [BCALM2](https://github.com/GATB/bcalm), [bifrost](https://github.com/pmelsted/bifrost)

**Comment by davidmaimoun** [comment]
Yes it will be smart to try all of these tools and compare the outputs

Thank you so much

**Comment by davidmaimoun** [comment]
Good Afternoon Dr Shen,
Could you explain me please what is the meaning of chunks in the kmcp output, and why we need it?
Thank you

**Comment by shenwei356** [comment]
[Search result format](https://bioinf.shenwei.me/kmcp/tutorial/profiling/#search-result-format): Tab-delimited format with 15 columns:

```
 1. query,    Identifier of the query sequence
 2. qLen,     Query length
 3. qKmers,   K-mer number of the query sequence
 4. FPR,      False positive rate of the match
 5. hits,     Number of matches
 6. target,   Identifier of the target sequence
 7. chunkIdx, Index of reference chunk
 8. chunks,   Number of reference chunks
 9. tLen,     Reference length
10. kSize,    K-mer size
11. mKmers,   Number of matched k-mers
12. qCov,     Query coverage,  equals to: mKmers / qKmers
13. tCov,     Target coverage, equals to: mKmers / K-mer number of reference chunk
14. jacc,     Jaccard index
15. queryIdx, Index of query sequence, only for merging
```

It's only used in [taxonomic profiling](https://doi.org/10.1101/2022.03.07.482835): 

> `kmcp compute` also supports splitting sequences into chunks, this could increase the specificity in profiling results at the cost of a slower searching speed.

See also Fig1a. 

![](https://github.com/shenwei356/kmcp/raw/main/kmcp.jpg)

**Comment by davidmaimoun** [comment]
It was very helpful

Thank you very much!
